anvil/anvil.go

```diff
@@ -109,11 +106,9 @@ func (a *Anvil) Start(ctx context.Context) error {
 		return fmt.Errorf("failed to start anvil: %w", err)
 	}
 
-	if _, err := utils.WaitForAnvilClientToBeReady(fmt.Sprintf("http://%s:%d", host, a.cfg.Port), 5*time.Second); err != nil {
-		return fmt.Errorf("failed to start anvil: %w", err)
-	}
-
 	go func() {
+		defer os.Remove(tempFile.Name())
```

Comment:
oh interesting! so moving `defer os.Remove(tempFile.Name())`, makes it so we no longer need to use `utils.WaitForAnvilClientToBeReady`? I'm curious since the call to `defer os.Remove(tempFile.Name())` is now here, is it possible that if the process errored out prior to this, would the `tempFile` not end up being removed?

Reply:
the behavior is slightly different than before 
- prev: as soon as anvil is ready we delete the temp file
- now: temp file is deleted when the process ends / errors out

mostly made this change so that the start function doesn't block - it returns immediately